esp-hal/src/uart.rs

```diff
@@ -2415,82 +2373,80 @@ impl Info {
         sync_regs(self.regs());
     }
 
-    #[cfg(any(esp32c6, esp32h2))]
     fn change_baud(&self, config: &Config) {
         let clocks = Clocks::get();
         let clk = match config.clock_source {
             ClockSource::Apb => clocks.apb_clock.as_hz(),
+            #[cfg(not(any(esp32, esp32s2)))]
             ClockSource::Xtal => clocks.xtal_clock.as_hz(),
+            #[cfg(not(any(esp32, esp32s2)))]
             ClockSource::RcFast => crate::soc::constants::RC_FAST_CLK.as_hz(),
+            #[cfg(any(esp32, esp32s2))]
+            ClockSource::RefTick => crate::soc::constants::REF_TICK.as_hz(),
         };
 
-        let max_div = 0b1111_1111_1111 - 1;
-        let clk_div = clk.div_ceil(max_div * config.baudrate);
-
-        // UART clocks are configured via PCR
-        let pcr = crate::peripherals::PCR::regs();
+        cfg_if::cfg_if! {
+            if #[cfg(any(esp32c2, esp32c3, esp32s3, esp32c6, esp32h2))] {
+
+                const MAX_DIV: u32 = 0b1111_1111_1111 - 1;
+                let clk_div = (clk.div_ceil(MAX_DIV)).div_ceil(config.baudrate);
+
+                // define `conf` in scope for modification below
+                cfg_if::cfg_if! {
+                    if #[cfg(any(esp32c2, esp32c3, esp32s3))] {
+                        if matches!(config.clock_source, ClockSource::RcFast) {
+                            crate::peripherals::LPWR::regs()
+                                .clk_conf()
+                                .modify(|_, w| w.dig_clk8m_en().variant(true));
+                            // esp_rom_delay_us(SOC_DELAY_RC_FAST_DIGI_SWITCH);
```

Comment:
Nitpick, I know it was already present but probably we don't need to keep this commented-out line?

Reply:
or maybe at some point (not here) we should do it - there are quite a few such delays in esp-idf

```c
/* Delays for various clock sources to be enabled/switched.
 * All values are in microseconds.
 */
#define SOC_DELAY_RTC_FAST_CLK_SWITCH       3
#define SOC_DELAY_RTC_SLOW_CLK_SWITCH       300
#define SOC_DELAY_RC_FAST_ENABLE            50
#define SOC_DELAY_RC_FAST_DIGI_SWITCH       5
#define SOC_DELAY_RC32K_ENABLE              300
```

Reply:
Thanks, I reworded the comment to something more useful instead of just commented out code.